cmd/exec/exec.go

```diff
@@ -1,69 +1,45 @@
 package exec
 
 import (
-	"errors"
+	"fmt"
+	"os"
+	"text/tabwriter"
 
-	"github.com/codeskyblue/go-sh"
+	sh "github.com/codeskyblue/go-sh"
 	"github.com/gopinath-langote/1build/cmd/config"
 	"github.com/gopinath-langote/1build/cmd/utils"
+	"github.com/logrusorgru/aurora"
 )
 
-// ExecuteCommands executes given command with before and after is present
-func ExecuteCommands(commands ...string) {
+// ExecutePlan executes the Execution plan
+func ExecutePlan(commands ...string) {
+
 	configuration, err := config.LoadOneBuildConfiguration()
 	if err != nil {
 		utils.PrintErr(err)
 		return
 	}
-	beforeCmd := configuration.Before
-	afterCmd := configuration.After
-
-	cmdMap, err := createMapOfCommandsToExecute(configuration, commands...)
-	if err != nil {
-		return
-	}
 
-	printMessage(cmdMap, beforeCmd, afterCmd)
+	executionPlan := buildExecutionPlan(configuration, commands...)
+	executionPlan.print()
 
-	if beforeCmd != "" {
-		err := executeAndStopIfFailed(beforeCmd)
-		if err != nil {
-			utils.Println("\nFailed to execute '" + beforeCmd + "'\n")
-			return
-		}
-	}
-	for _, v := range cmdMap {
-		err = executeAndStopIfFailed(v)
-		if err != nil {
-			utils.Println("\nFailed to execute '" + v + "'\n")
-			return
-		}
+	if executionPlan.hasInit() {
```

Comment:
For better readability - can we call the method as `executionPlan.hasBefore()` and `executionPlan.hasAfter()` - as we call them in yaml config?

WDYT?

Reply:
Maybe. But when we build an execution plan `init` and `destroy` makes more sense. `before` and `after` can be used more like before and after every command.

WDYT?